build/v1/types.go

```diff
@@ -661,6 +661,21 @@ const (
 	// warn to the build output instead of failing when unsupported syntax is detected. This
 	// policy is experimental.
 	ImageOptimizationSkipLayersAndWarn ImageOptimizationPolicy = "SkipLayersAndWarn"
+
+	// ImageOptimizationDaemonless is an experimental policy which builds an image without
+	// using a docker engine, typically by running a build tool directly inside of the
+	// builder container, using a single layer for all content changes relative to the base
+	// image.
+	ImageOptimizationDaemonless ImageOptimizationPolicy = "Daemonless"
```

Comment:
This isn't the appropriate place to put this.  None of these have anything to do with ImageOptimization.  The pod should detect whether to use daemonless or not by whether the docker socket is present or another setting on the pod is passed in.

End users shouldn't be specifying this.

Reply:
I'm not wedded to using the ImageOptimizationPolicy field to control whether or not we use buildah for the build.  At least for making things easier for myself, I'd like to be able to directly control whether or not we opt in to exercising new logic.  Would something like adding an `ExposeDockerSocket` boolean (perhaps reversed to default to the current behavior when unset) to DockerBuildStrategy/SourceBuildStrategy work better?

Reply:
> Would something like adding an ExposeDockerSocket boolean (perhaps reversed to default to the current behavior when unset) to DockerBuildStrategy/SourceBuildStrategy work better?

i would say no.

At this point i think we are heading down a path where we just always use buildah and at least for 4.0 i'm inclined to say we pick a single layering strategy (probably "add a single new layer on top of your base image layers").